DatadogRUM/Sources/Integrations/TelemetryReceiver.swift

```diff
@@ -254,6 +254,30 @@ internal final class TelemetryReceiver: FeatureMessageReceiver {
     }
 }
 
+fileprivate extension [String: Encodable] {
+    func extended(
+        with context: DatadogContext
+    ) -> [String: Encodable] {
+        /// Method Called metric needs to be enriched with device and os information, only available in the `DatadogContext`.
+        if let key = self[BasicMetric.typeKey] as? String, key == MethodCalledMetric.typeValue {
```

Comment:
I was trying to prepare for future use cases where we potentially want to enrich with some context data.

One way to prettify it would be some extension functions that can tell if dictionary if representing attributes of given metric type.